### PR TITLE
refactor, chore, feat(egen): contracts content, steps & template based fixes

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
The notebook `hyperparameter_tuning_xgboost.ipynb` consists of some elaborate explanations and steps which can be simplified.
This PR includes the following updates:

1. Restructures the notebook according to the template.
2. Simplifies elaborative/unnecessary steps and content.
3. Replaces K80 with T4.
4. Reduces the disk space from 200 to 100 (to set a minimum).
5. Removes future tenses and simplifies wording.
6. Adds colab enterprise link.
7. Simplifies the section headings
8. Removes redundant code/explanations.
9. Adds clean up step for locally generated files.
10. Removes -q in the installation step (--quiet is already available).
11. Replaces REGION with LOCATION.
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 1]
```diff
@@ -74,21 +76,19 @@
       "source": [
         "### Objective\n",
         "\n",
-        "In this tutorial, you learn how to use `Vertex AI Hyperparameter Tuning` for training a XGBoost custom model.\n",
+        "In this tutorial, you learn how to use **Vertex AI Hyperparameter Tuning** service for training an XGBoost model.\n",
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:79]: should be:
- Vertex AI hyperparameter tuning

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 2]
```diff
@@ -74,21 +76,19 @@
...
         "\n",
-        "This tutorial uses the following Google Cloud ML services:\n",
+        "This tutorial uses the following Vertex AI services:\n",
         "\n",
-        "- `Vertex AI Training`\n",
-        "- `Vertex AI Hyperparameter Tuning`\n",
-        "- `Vertex AI Vizier`\n",
-        "- `Vertex AI Model` resource\n",
+        "- **Vertex AI Training**\n",
+        "- **Vertex AI Hyperparameter Tuning** (uses **Vertex AI Vizier**)\n",
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:84]: again, should be:
- Vertex AI hyperparameter tuning

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 3]
```diff
@@ -477,38 +403,19 @@
...
       "cell_type": "markdown",
       "metadata": {
         "id": "examine_training_package:xgboost"
       },
       "source": [
-        "### Examine the training package\n",
+        "## Python training application package\n",
         "\n",
-        "#### Package layout\n",
+        "In this example, you use Vertex AI Hyperparameter Tuning service with a training job that executes a Python training application package.\n",
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:414]: Should be:
- Vertex AI hyperparameter tuning

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 4]
```diff
@@ -619,9 +531,8 @@
         "\n",
         "def get_data():\n",
         "    logging.info(\"Downloading training data and labelsfrom: {}, {}\".format(args.dataset_data_url, args.dataset_labels_url))\n",
-        "    # gsutil outputs everything to stderr so we need to divert it to stdout.\n",
+        "    # gsutil outputs everything to stderr. Hence, need to divert it to stdout.\n",
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:534]: change to:
- Hence, the need to divert it to ...

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 5]
```diff
@@ -692,9 +603,9 @@
         "id": "tarball_training_script"
       },
       "source": [
-        "#### Store training script on your Cloud Storage bucket\n",
+        "### Store training script on Cloud Storage bucket\n",
         "\n",
-        "Next, you package the training folder into a compressed tar ball, and then store it in your Cloud Storage bucket."
+        "Compress the whole training folder as a tar ball and then store it in Cloud Storage bucket."
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:608]: Add "a"
- "store it in a Cloud Storage bucket

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 6]
```diff
@@ -903,7 +818,7 @@
       "source": [
         "## Run the hyperparameter tuning job\n",
         "\n",
-        "Use the `run()` method to execute the hyperparameter tuning job."
+        "Use `run()` method to execute the hyperparameter tuning job."
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:821]: put "the" back in.

File: notebooks/official/training/hyperparameter_tuning_xgboost.ipynb [hunk 7]
```diff
@@ -925,7 +840,7 @@
       "source": [
         "### Display the hyperparameter tuning job trial results\n",
         "\n",
-        "After the hyperparameter tuning job has completed, the property `trials` will return the results for each trial."
+        "Once the hyperparameter tuning job gets finished successfully, you can access the results from each trial using the `trials` property."
```
Comment (kittyabs) [inline notebooks/official/training/hyperparameter_tuning_xgboost.ipynb:843]: change to: 
- "Once the hyperparameter tuning job successfully finishes, you can...

**Review by kittyabs — COMMENTED** [review]
Made several suggested edits

**Review by katiemn — APPROVED** [review]

**Pull request merged**